### PR TITLE
fix: release reserved runtime push tokens

### LINKED ISSUE #115: Add runtime push token reclamation and protocol coverage
## Problem

PR #114 introduced broker-owned `push.reserve` tokens so runtime listeners can reserve a push token before a base agent session exists. Reviewers agreed this is the right architecture, but it leaves follow-up design debt:

- reserved-only push tokens persist until broker restart
- `DeletePushToken` has no protocol-level production caller
- protocol-level idempotency and already-connected-agent behavior are not pinned by tests
- broker-restart re-reservation is not directly covered

This is not a PR #114 merge blocker, but it should be closed while the context is fresh.

## Acceptance criteria

- Define and implement an explicit reclamation path for broker-owned runtime push tokens, or document and test why broker-lifetime tokens are intentional.
- Add protocol-level tests for duplicate `push.reserve` calls returning the same token.
- Add protocol-level tests for `push.reserve` when the base agent is already connected.
- Add lifecycle coverage showing the intended listener behavior remains intact across base-agent reconnect/disconnect cycles.
- If a protocol command is added, update `internal/protocol` tests and keep startup/runtime paths degraded-safe.

## Notes

PR #114 head reviewed: `b14417979533efaf712ac759a794bbee0eaab4c3`
Merged as: `90b92d8e9a4395a64079e4a5d4dfd78d95e15e13`

### PR DESCRIPTION
**Pull request by seungpyoson**
## Summary

- Add `push.release` to reclaim broker-owned runtime push tokens by base name and token.
- Have broker runtime listeners release their reserved token on listener shutdown instead of leaving reserved-only tokens until broker restart.
- Add protocol/lifecycle coverage for duplicate reserves, reserve while base is connected, release validation, and repeated base reconnects.

Closes #115.

## Verification

- `go test ./... -count=1 -timeout=120s`
- `go test ./internal/broker ./internal/runtime ./internal/protocol -count=1 -timeout=60s`
- `go build -o waggle .`
- `git diff --check`

### CODE REVIEW

File: internal/broker/router.go [hunk 1]
```diff
@@ -114,6 +117,38 @@ func handlePushReserve(s *Session, req protocol.Request) protocol.Response {
...
+		return protocol.ErrResponse(protocol.ErrForbidden, "invalid push listener token")
+	}
+	delete(s.broker.pushTokens, req.Name)
+	pushListener = s.broker.sessions[req.Name+"-push"]
+	s.broker.mu.Unlock()
+
+	if pushListener != nil {
+		pushListener.cleanDisconnect = true
+		pushListener.cleanup()
+	}
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply internal/broker/router.go]: Addressed by removing the active listener cleanup path from `handlePushRelease` entirely. `push.release` now only deletes the token and clears stale base-session ownership, so it cannot close the caller's connection or any fresh `name-push` session before a response is encoded.

File: internal/runtime/listener.go [hunk 2]
```diff
@@ -135,7 +135,10 @@ func (l *brokerListener) Listen(ctx context.Context, handler DeliveryHandler) er
 	if err != nil {
 		return err
 	}
-	defer disconnectClient(c)
+	defer func() {
+		c.Close()
+		releasePushTokenForAgent(l.socketPath, l.agentName, pushToken)
+	}()
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply internal/runtime/listener.go:155]: Addressed. `Listen` now releases the reserved token if the post-reserve broker dial fails, and its cleanup defer is structured immediately after the listener connection is established. The defer releases on transport/setup failure or after a successful listener session shuts down, but deliberately disables release for protocol handshake rejection such as `ALREADY_CONNECTED` so a failed duplicate listener cannot revoke the active listener's token.

File: internal/runtime/listener.go [hunk 3]
```diff
@@ -247,3 +250,32 @@ func pushTokenForAgent(socketPath, agent string) (string, error) {
 	}
 	return data.PushToken, nil
 }
+
+func releasePushTokenForAgent(socketPath, agent, pushToken string) {
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply internal/runtime/listener.go:269]: Addressed. `CatchUp` now releases the reserved token after a successful catch-up session, and releases on post-reserve dial failure. Added `TestBrokerListenerCatchUpReleasesPushToken` to prove the token cannot be reused after catch-up completes.
